### PR TITLE
github actions ci

### LINKED ISSUE #555: CI: migrate to github actions
Travis seems to be disabled on this repo. We should migrate to github actions and get rid of appveyor as well.

### PR DESCRIPTION
**Pull request by ordian**
Fixes #555.

Doesn't include tests for kvdb-web, I'm planning to decommission it soon.
- [x] update readme badge
- [x] test parity-util-mem allocators

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -0,0 +1,153 @@
...
+      - uses: actions/checkout@v2
+      - uses: actions-rs/toolchain@v1
+        with:
+          profile: minimal
+          toolchain: stable
+          override: true
+      - uses: actions-rs/cargo@v1
+        with:
+          command: check
+          args: --workspace --benches
```
Comment (niklasad1) [inline .github/workflows/ci.yml:23]: why not `--all-targets`? or the test builds are caught later by the tests?

**Review by dvdplm — APPROVED** [review]

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -0,0 +1,153 @@
+on:
+  pull_request:
+  push:
+    branches:
+      - master
+
+name: Continuous integration
+
+jobs:
```
[the remark this answers is not in the document]
Comment (TriplEight) [reply .github/workflows/ci.yml:9]: didn't do proper benchmarks, but it covers more cache than the native cache action.

File: .github/workflows/ci.yml [hunk 3]
```diff
@@ -0,0 +1,156 @@
...
+          args: --workspace
+
+      - name: Test uint
+        uses: actions-rs/cargo@v1
+        with:
+          command: test
+          args: -p uint --all-features
+
+      - name: Test fixed-hash no_std
+        run: cargo test -p fixed-hash --no-default-features --features='byteorder,rustc-hex'
```
Comment (TriplEight) [inline .github/workflows/ci.yml:74]: curious: what stopped you from using the `actions-rs/cargo@v1` here?
(I personally think this way is more readable)

**Review by TriplEight — APPROVED** [review]
everything's honky-dory!

File: .github/workflows/ci.yml [hunk 4]
```diff
@@ -0,0 +1,156 @@
...
+    name: Check
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+      - uses: actions-rs/toolchain@v1
+        with:
+          profile: minimal
+          toolchain: stable
+          override: true
+      - uses: actions-rs/cargo@v1
```
Comment (TriplEight) [inline .github/workflows/ci.yml:20]:
```suggestion

      - name: Rust Cache
        uses: Swatinem/rust-cache@v1.3.0

      - uses: actions-rs/cargo@v1
```

File: .github/workflows/ci.yml (hunk 3 above)
Comment (ordian) [reply .github/workflows/ci.yml:74]: it didn't work as you can see from the commit history and CI status
https://github.com/paritytech/parity-common/runs/2954854743
https://github.com/paritytech/parity-common/runs/2956324167

File: .github/workflows/ci.yml [hunk 5]
```diff
@@ -0,0 +1,156 @@
...
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v1
+      - uses: actions-rs/toolchain@v1
+        with:
+          profile: minimal
+          toolchain: stable
+          override: true
+      - run: rustup component add rustfmt
+      - uses: actions-rs/cargo@v1
```
Comment (TriplEight) [inline .github/workflows/ci.yml:153]:
```suggestion

      - name: Rust Cache
        uses: Swatinem/rust-cache@v1.3.0

      - uses: actions-rs/cargo@v1
```

Comment (ordian) [reply .github/workflows/ci.yml:153]: I don't think we need a cache for cargo fmt or cargo check

**Comment by TriplEight** [comment]
I do not know how loaded this project is, so please tell me if these pipelines are taking too much time for your taste, then we'll rewrite some parts on GitLab, it would be a way faster.

File: .github/workflows/ci.yml (hunk 5 above)
Comment (TriplEight) [reply .github/workflows/ci.yml:153]: yeah, it takes 20 seconds anyway

**Comment by ordian** [comment]
> I do not know how loaded this project is, so please tell me if these pipelines are taking too much time for your taste, then we'll rewrite some parts on GitLab, it would be a way faster.

It's quite slow w/o a cache, up to 30 mins, but we can live with that for now and hopefully cache will help as well.

File: .github/workflows/ci.yml (hunk 3 above)
Comment (TriplEight) [reply .github/workflows/ci.yml:74]: meh, classic

**Comment by TriplEight** [comment]
> It's quite slow w/o a cache, up to 30 mins, but we can live with that for now and hopefully cache will help as well.

It will definitely help.
But there's a limit of 5Gb of cache per project and if it's more than that, that's a sign to move to GitLab CI.

**Comment by TriplEight** [comment]
image

This looks a way better than this:

image

**Pull request merged**